### PR TITLE
Feature/devcred 654 refac merchant addresses

### PR DESCRIPTION
**Pull request by caiomdias**
## Proposed Changes

Refactor the affiliation-merchant-addresses component for the new architecture.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] All tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further Comments

### CODE REVIEW

File: packages/affiliation-merchant-addresses/src/component/MerchantAddresses.js [hunk 1]
```diff
@@ -50,45 +29,37 @@ export class AffiliationMerchantAddresses extends SlingBusinessElement {
...
-    `;
+  static get requestParamNames() {
+    return ['affiliationCode'];
+  }
+
+  fetchData({ affiliationCode }) {
+    if (affiliationCode) {
+      this
+        .request([
+          SDK.merchants.addresses.get({ affiliationCode }),
```
[the remark this answers is not in the document]
Comment (caiomdias) [reply packages/affiliation-merchant-addresses/src/component/MerchantAddresses.js]: O `merchants` significa modulo de merchants, que contempla os endpoints da afiliação na SDK, eu estou querendo mudar pra affiliation mas isso iria ocasionar em um refac enorme.

File: packages/affiliation-merchant-addresses/src/component/MerchantAddresses.js [hunk 2]
```diff
@@ -1,47 +1,26 @@
...
-import 'sling-web-component-form';
-import 'sling-web-component-input';
-import 'sling-web-component-select';
-import 'sling-web-component-loader';
-import 'sling-web-component-message';
-import 'sling-web-component-button';
+import { v0 as SDK } from 'customer-js-sdk';
+import { withRequest, withSetState } from 'sling-framework';
+import { addressesModel } from '../model/MerchantAddressesModel.js';
+import { MerchantAddressesView } from '../view/MerchantAddressesView.js';
```
[the remark this answers is not in the document]
Comment (caiomdias) [reply packages/affiliation-merchant-addresses/src/component/MerchantAddresses.js]: Estou mudando, eu fiz errado mesmo 💩

File: packages/affiliation-merchant-addresses/src/component/MerchantAddresses.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (psvaiter) [reply packages/affiliation-merchant-addresses/src/component/MerchantAddresses.js]: Acho justo um substantivo que represente o domínio, como você sugeriu, em vez de colocar um nome de entidade no plural. Acho válida uma refatoração nesse sentido.

**Pull request merged**